vars/slingOsgiBundleBuild.groovy

```diff
@@ -63,7 +63,20 @@ def call(Map params = [:]) {
 
                     if ( jobConfig.sonarCloud ) {
                         stage('SonarCloud') {
-                            echo "stub"
+                            // As we don't have the global SonarCloud conf for now, we can't use #withSonarQubeEnv so we need to set the following props manually
+                            def sonarcloudParams="-Dsonar.host.url=https://sonarcloud.io -Dsonar.organization=apache"
+                            // TODO below: replace SC_TOKEN by the right way to get Robert's SonarCloud token, probably using #withCredentials
+                            sonarcloudParams="${sonarcloudParams} -Dsonar.login=${SC_TOKEN}"
+                            // Params are different if it's a PR or if it's not
+                            // Note: soon we won't have to handle that manually, see https://jira.sonarsource.com/browse/SONAR-11853
+                            if ( isPrBuild ) {
+                                sonarcloudParams="${sonarcloudParams} -Dsonar.pullrequest.branch=${CHANGE_BRANCH} -Dsonar.pullrequest.base=${CHANGE_TARGET} -Dsonar.pullrequest.key=${CHANGE_ID}"
+                            } else {
+                                sonarcloudParams="${sonarcloudParams} -Dsonar.branch.name=${BRANCH_NAME}"
+                            }
+                            // Alls params are set, let' execute
+                            def mvnCmd = "mvn -U clean verify sonar:sonar ${sonarcloudParams}"
```

Comment:
Thanks for the PR @bellingard ! I'll add the token to the execution, but was wondering whether we need to use the trick from https://github.com/apache/sling-tooling-jenkins/blob/eb00228b2e85132058adde6561d00afea20cc3b2/vars/slingOsgiBundleBuild.groovy#L51-L53 to prevent the `sonar.login` token from being printed in the Jenkins build log?

Reply:
I've improved the code to wrap the command using withCredentials, this will mask the token if it is printed out.
Also, note that I've used a "secret text" type credential (named "robert_token" in this example). This is what you should set on the Sling folder in Jenkins.

Reply:
Thanks @bellingard . I'll merge this as-is and do any necessary tweaks later.